crates/sui-graphql-client/src/lib.rs

```diff
@@ -683,6 +688,82 @@ impl Client {
         }
     }
 
+    // ===========================================================================
+    // Dry Run API
+    // ===========================================================================
+
+    /// Dry run a [`Transaction`] and return the transaction.
+    ///
+    /// `skipChecks` optional flag disables the usual verification checks that prevent access to
+    /// objects that are owned by addresses other than the sender, and calling non-public,
+    /// non-entry functions, and some other checks. Defaults to false.
+    pub async fn dry_run_tx(
+        &self,
+        tx: &Transaction,
+        skip_checks: Option<bool>,
+    ) -> Result<Option<Transaction>, Error> {
```

Comment:
This return type doesn't make any sense. If i'm performing a dryrun i want to preview the effects of the transaction, not get back the exact txn that i sent

Reply:
🤦  yes, you're right.